Source/Csla/Core/BusyHelper.cs

```diff
@@ -0,0 +1,49 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="INotifyBusy.cs" company="Marimer LLC">
+//     Copyright (c) Marimer LLC. All rights reserved.
+//     Website: https://cslanet.com
+// </copyright>
+// <summary>Interface defining an object that notifies when it</summary>
+//-----------------------------------------------------------------------
+using System;
+using System.Runtime.CompilerServices;
+using System.Threading.Tasks;
+
+namespace Csla.Core {
+  /// <summary>
+  /// Helper class for busy related functionality spread across different business type implementations.
+  /// </summary>
+  internal static class BusyHelper {
+    public static async Task WaitForIdle(INotifyBusy source, TimeSpan timeout, [CallerMemberName] string methodName = "") {
+      if (!source.IsBusy) {
+        return;
+      }
+
+      var tcs = new TaskCompletionSource<object>();
+      try {
+        source.BusyChanged += ObserverForIsBusyChange;
+
+        if (!source.IsBusy) {
+          return;
+        }
+
+        var timeoutTask = Task.Delay(timeout);
+        var finishedTask = await Task.WhenAny(tcs.Task, timeoutTask);
+
+        if (finishedTask == timeoutTask) {
+          throw new TimeoutException($"{source.GetType().FullName}.{methodName} after {timeout}.");
```

Comment:
If we put actual words in an exception message, that message needs to come from the resources (resx) file so it can be localized.

Lately I've been trying hard to avoid the use of any words in exceptions to minimize the localization effort in that regard.

Reply:
That's a good approach to avoid that nasty thing called localization :D

Reply:
So instead of the "after" just a "-" (dash) to include the timeout time or should I remove that completly and do it like the old implementation did?

Reply:
The dash works for me.

Reply:
Done